connectors/src/connectors/intercom/temporal/sync_help_center.ts

```diff
@@ -0,0 +1,346 @@
+        loggerArgs,
+      });
+    })
+  );
+}
+
+/**
+ * Syncs a collection and its children (collection & articles) from the database and the core data source.
+ */
+export async function _upsertCollection({
+  connectorId,
+  intercomClient,
+  dataSourceConfig,
+  loggerArgs,
+  collection,
+  parents,
+  currentSyncMs,
+}: {
+  connectorId: ModelId;
+  intercomClient: IntercomClient;
+  dataSourceConfig: DataSourceConfig;
+  collection: IntercomCollectionType;
+  loggerArgs: Record<string, string | number>;
+  parents: string[];
+  currentSyncMs: number;
+}) {
+  // Sync the Collection
+  let collectionOnDb = await IntercomCollection.findOne({
+    where: {
+      connectorId,
+      collectionId: collection.id,
+    },
+  });
+
+  if (collectionOnDb) {
+    await collectionOnDb.update({
+      name: collection.name,
+      description: collection.description,
+      lastUpsertedTs: new Date(currentSyncMs),
+    });
+  } else {
+    collectionOnDb = await IntercomCollection.create({
+      connectorId: connectorId,
+      collectionId: collection.id,
+      intercomWorkspaceId: collection.workspace_id,
+      helpCenterId: collection.help_center_id,
+      parentId: collection.parent_id,
+      name: collection.name,
+      description: collection.description,
+      url: collection.url,
+      permission: "read",
+      lastUpsertedTs: new Date(currentSyncMs),
+    });
+  }
+
+  // Sync the Collection's articles
+  const [childrenArticlesOnIntercom, childrenArticlesOnDb] = await Promise.all([
+    fetchIntercomArticles(intercomClient, collection.id),
+    IntercomArticle.findAll({
+      where: { connectorId, parentId: collection.id },
+    }),
+  ]);
+
+  const promises = childrenArticlesOnIntercom.map(async (articleOnIntercom) => {
+    const matchingArticleOnDb = childrenArticlesOnDb.find(
+      (article) => article.articleId === articleOnIntercom.id
+    );
+    let article = null;
+    if (matchingArticleOnDb) {
+      article = await matchingArticleOnDb.update({
+        title: articleOnIntercom.title,
+        url: articleOnIntercom.url,
+        lastUpsertedTs: new Date(currentSyncMs),
+      });
+    } else {
+      article = await IntercomArticle.create({
+        connectorId: connectorId,
+        articleId: articleOnIntercom.id,
+        title: articleOnIntercom.title,
+        url: articleOnIntercom.url,
+        intercomWorkspaceId: articleOnIntercom.workspace_id,
+        authorId: articleOnIntercom.author_id,
+        parentId: articleOnIntercom.parent_id,
+        parentType:
+          articleOnIntercom.parent_type === "collection" ? "collection" : null,
+        parents: articleOnIntercom.parent_ids,
+        state: articleOnIntercom.state === "published" ? "published" : "draft",
+        permission: "read",
+        lastUpsertedTs: new Date(currentSyncMs),
+      });
+    }
+
+    const content =
+      "CATEGORY: " +
+      collection.description +
+      "\nCONTENT: " +
+      articleOnIntercom.body;
```

Comment:
Is that body anything? markdown?

Reply:
It was HTML, thanks for catching. I added a commit to upsert is as text